waku/v2/node/wakunode2.nim

```diff
@@ -468,26 +468,29 @@ proc mountStore*(node: WakuNode, store: MessageStore = nil, persistMessages: boo
   node.switch.mount(node.wakuStore, protocolMatcher(WakuStoreCodec))
     
 when defined(rln):
-  proc addRLNRelayValidator*(node: WakuNode, pubsubTopic: string) =
+  proc addRLNRelayValidator*(node: WakuNode, pubsubTopic: string, contentTopic: ContentTopic) =
     ## this procedure is a thin wrapper for the pubsub addValidator method
     ## it sets message validator on the given pubsubTopic, the validator will check that
     ## all the messages published in the pubsubTopic have a valid zero-knowledge proof 
     proc validator(topic: string, message: messages.Message): Future[pubsub.ValidationResult] {.async.} =
       let msg = WakuMessage.init(message.data) 
       if msg.isOk():
-        let 
-          wakumessage = msg.value()
-          # validate the message
-          validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
+        let wakumessage = msg.value()
+        # check the contentTopic
+        if (wakumessage.contentTopic != "") and (wakumessage.contentTopic != contentTopic):
+          trace "content topic did not match:", contentTopic=wakumessage.contentTopic, payload=string.fromBytes(wakumessage.payload)
+          return pubsub.ValidationResult.Accept
+        # validate the message
+        let validationRes = node.wakuRlnRelay.validateMessage(wakumessage)
         case validationRes:
           of Valid:
-            info "message validity is verified, relaying:", wakumessage=wakumessage
+            trace "message validity is verified, relaying:", wakumessage=wakumessage, payload=string.fromBytes(wakumessage.payload)
```

Comment:
Would it make sense to add a metric here that counts all the different types of validation responses?

Reply:
Thanks, for the metrics, I am gonna open another PR in which I will also include the validation responses.